book/src/proptest-derive/modifiers.md

```diff
@@ -0,0 +1,370 @@
+# Modifier Reference
+
+All modifiers interpreted by `#[derive(Arbitrary)]` are of the form
+`#[proptest(..)]`, where the content between the parentheses follows the normal
+Rust attribute syntax.
+
+Each modifier within the parentheses is independent, in that putting two
+modifiers in the same attribute is equivalent to having two `#[proptest(..)]`
+attributes with one modifier each.
+
+For brevity, modifiers are sometimes referenced by name alone; e.g., "the
+`weight` modifier" refers to `#[proptest(weight = nn)]` and not some
+freestanding `#[weight]` attribute.
+
+## `filter`
+
+Form: `#[proptest(filter = F)]` or `#[proptest(filter(F))]` where `F` is either
+a bare identifier or a Rust expression in a string. In either case, the
+parameter must evaluate to something which is `Fn (&T) -> bool`, where `T` is
+the type of the item being filtered.
+
+Usable on: structs, enums, enum variants, fields
+
+The `filter` modifier allows filtering values generated for a field via
+rejection sampling. Since rejection sampling is inefficient and interferes with
+shrinking, it should only be used for conditions that are very rare or are
+unfeasible to express otherwise. In many cases, [`strategy`](#strategy) can be
+used to more directly express the desired behaviour without rejection sampling.
+See the documentation for [`prop_filter`] for more details.
+
+The argument to the modifier must be a valid argument for the second parameter
+of [`prop_filter`].
+
+Example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+#[proptest(filter = "|segment| segment.start != segment.end")]
+struct NonEmptySegment {
+    start: i32,
+    end: i32,
+}
+```
+
+As mentioned above, filtering should be avoided when it is reasonably possible
+to express a non-filtering strategy that achieves the same effect. For example:
+
+```rust
+#[derive(Debug, Arbitrary)]
+struct BadExapmle {
+    // Don't do this! Your tests will run more slowly and shrinking won't work
+    // properly.
+    #[proptest(filter = "|x| x % 2 == 0")]
```

Comment:
Maybe extract this out to an `is_even` function to demonstrate how that works?

Reply:
I added a separate example to the main usage example.